server/commands/cycle.js

```diff
@@ -1,47 +1,64 @@
 /* global graphQLFetcher, notifyUser, commandsConfig */
 
-function invokeLaunchCycleAPI(lgJWT) {
+function invokeUpdateCycleStateAPI(state, lgJWT) {
   const baseURL = process.env.NODE_ENV === 'development' ? 'http://game.learnersguild.dev' : 'https://game.learnersguild.org'
   const mutation = {
-    query: 'mutation { launchCycle { id } }'
+    query: 'mutation($state: String!) { updateCycleState(state: $state) { id } }',
+    variables: {state},
   }
   return graphQLFetcher(lgJWT, baseURL)(mutation)
     .then(data => data.launchCycle)
 }
 
-function launchCycle(commandInfo) {
+function handleUpdateCycleStateCommand(commandInfo, state, msg) {
   try {
     const {lgJWT, lgPlayer} = Meteor.user().services.lgSSO
     if (!lgJWT || !lgPlayer) {
       throw new Error('You are not a player in the game.')
     }
-    notifyUser(commandInfo.rid, '🚀  Initiating Launch... stand by.')
-    invokeLaunchCycleAPI(lgJWT)
-      // unless our API invocation fails, we'll stay quiet because the server will notify
-      // the user via the web socket
-      // .then(response => console.log(`[LG SLASH COMMANDS] '/cycle launch' API response: ${response}`))
+    notifyUser(commandInfo.rid, msg)
+    invokeUpdateCycleStateAPI(state, lgJWT)
       .catch(error => {
         console.error(error.stack)
         RavenLogger.log(error)
-        notifyUser(commandInfo.rid, '**FATAL**: API invocation failed.')
+        notifyUser(commandInfo.rid, `**ERROR**: ${error.message}`)
       })
   } catch (errorMessage) {
     notifyUser(commandInfo.rid, `**ERROR:** ${errorMessage.message}`)
```

Comment:
Am I mistaken or would this try/catch only catch the one error explicitly thrown in the case of user authorization failure (line 17)?

Would it be clearer/make more sense to only have that little bit wrapped by the try/catch?

Or is there an intention for this to capture something thrown by the async `invokeUpdateCycleStateAPI()` call?